src/main/java/io/antmedia/datastore/db/MongoStore.java

```diff
@@ -1429,4 +1432,103 @@ public boolean updateStreamMetaData(String streamId, String metaData) {
 		}
 		return false;
 	}
+
+	@Override
+	public boolean blackListToken(Token token) {
+		boolean result = false;
+		//update if exists, else insert
+		synchronized (this) {
```

Comment:
The scope of the synchronized block looks unnecessarily broad and blocks other concurrent operations in the store. Sorry if I misunderstand, but if you are trying to ensure some kind of atomic transaction here so that the `find-update-save` cannot be interrupted, then what about other JVM participants in the cluster, that are not held up by the lock, hence can still concurrently fiddle with Mongo?